#Netflix Data Analysis
###Exploratory Data Analysis of Netflix movies and TV shows dataset and drawing some insights/conclusions from it.

In [ ]:
# Import libraries for data analysis and visualization

import pandas as pd           # For data manipulation
import plotly.express as px   # For interactive visualizations

## Data Loading

In [ ]:
#In this step we load the dataset into a Pandas DataFrame.
df = pd.read_csv('/content/NetflixData.csv')

## Data Preview

In [ ]:
# Show first 5 rows; if you write head(10) it will display first 10 rows, change the number inside head() to display that number of rows.
df.head()

,Release_Date,Title,Overview,Popularity,Vote_Count,Vote_Average,Original_Language,Genre,Poster_Url
0,15-12-2021,Spider-Man: No Way Home,Peter Parker is unmasked and no longer able to...,5083.954,8940,8.3,en,"Action, Adventure, Science Fiction",https://image.tmdb.org/t/p/original/1g0dhYtq4i...
1,01-03-2022,The Batman,"In his second year of fighting crime, Batman u...",3827.658,1151,8.1,en,"Crime, Mystery, Thriller",https://image.tmdb.org/t/p/original/74xTEgt7R3...
2,25-02-2022,No Exit,Stranded at a rest stop in the mountains durin...,2618.087,122,6.3,en,Thriller,https://image.tmdb.org/t/p/original/vDHsLnOWKl...
3,24-11-2021,Encanto,"The tale of an extraordinary family, the Madri...",2402.201,5076,7.7,en,"Animation, Comedy, Family, Fantasy",https://image.tmdb.org/t/p/original/4j0PNHkMr5...
4,22-12-2021,The King's Man,As a collection of history's worst tyrants and...,1895.511,1793,7.0,en,"Action, Adventure, Thriller, War",https://image.tmdb.org/t/p/original/aq4Pwv5Xeu...


## Dataset Structure

In [ ]:
# Show number of rows and columns
df.shape

(9826, 9)

In [ ]:
# Show column names
df.columns

Index(['Release_Date', 'Title', 'Overview', 'Popularity', 'Vote_Count',
       'Vote_Average', 'Original_Language', 'Genre', 'Poster_Url'],
      dtype='object')

In [ ]:
# Show dataset information
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 9826 entries, 0 to 9825
Data columns (total 9 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   Release_Date       9826 non-null   object 
 1   Title              9825 non-null   object 
 2   Overview           9826 non-null   object 
 3   Popularity         9825 non-null   float64
 4   Vote_Count         9826 non-null   int64  
 5   Vote_Average       9825 non-null   float64
 6   Original_Language  9826 non-null   object 
 7   Genre              9825 non-null   object 
 8   Poster_Url         9826 non-null   object 
dtypes: float64(2), int64(1), object(6)
memory usage: 691.0+ KB


## Statistical Summary

In [ ]:
# Show statistical summary of numerical columns
df.describe()

,Popularity,Vote_Count,Vote_Average
count,9825.000000,9826.000000,9825.000000
mean,40.502234,1392.749746,6.439277
std,110.565717,2611.352214,1.129797
min,13.354000,0.000000,0.000000
25%,16.128000,146.000000,5.900000
50%,21.191000,444.000000,6.500000
75%,35.165000,1375.750000,7.100000
max,5083.954000,31077.000000,10.000000


## Data Quality Check

In [ ]:
# Check duplicate rows
df.duplicated()

,0
0,False
1,False
2,False
3,False
4,False
...,...
9821,False
9822,False
9823,False
9824,False


In [ ]:
# Count duplicate rows
df.duplicated().sum()

np.int64(2)

In [ ]:
# Count missing values in each column
df.isnull().sum()

,0
Release_Date,0
Title,1
Overview,0
Popularity,1
Vote_Count,0
Vote_Average,1
Original_Language,0
Genre,1
Poster_Url,0


## Data Cleaning

In [ ]:
# Remove duplicate rows
df = df.drop_duplicates()

# Confirm duplicates removed
df.duplicated().sum()

np.int64(0)

In [ ]:
# Remove rows with missing values
df = df.dropna()

# Confirm missing values removed
df.isnull().sum()

,0
Release_Date,0
Title,0
Overview,0
Popularity,0
Vote_Count,0
Vote_Average,0
Original_Language,0
Genre,0
Poster_Url,0


## Data Preprocessing / Data Transformation

In [ ]:
# Convert Release_Date to datetime format
df['Release_Date'] = pd.to_datetime(df['Release_Date'])

# Extract year from release date
df['Release_Date'] = df['Release_Date'].dt.year

# Display sample row
df.head(1)

/tmp/ipykernel_294/3326588578.py:2: UserWarning: Parsing dates in %d-%m-%Y format when dayfirst=False (the default) was specified. Pass `dayfirst=True` or specify a format to silence this warning.
  df['Release_Date'] = pd.to_datetime(df['Release_Date'])


,Release_Date,Title,Overview,Popularity,Vote_Count,Vote_Average,Original_Language,Genre,Poster_Url
0,2021,Spider-Man: No Way Home,Peter Parker is unmasked and no longer able to...,5083.954,8940,8.3,en,"Action, Adventure, Science Fiction",https://image.tmdb.org/t/p/original/1g0dhYtq4i...


In [ ]:
# Drop columns not needed for analysis
df = df.drop(columns=['Overview','Poster_Url','Original_Language'])

In [ ]:
# Convert genre string into list
df['Genre'] = df['Genre'].str.split(', ')

# Split genres into multiple rows
df = df.explode('Genre').reset_index(drop=True)

# Show first rows
df.head()

,Release_Date,Title,Popularity,Vote_Count,Vote_Average,Genre
0,2021,Spider-Man: No Way Home,5083.954,8940,8.3,Action
1,2021,Spider-Man: No Way Home,5083.954,8940,8.3,Adventure
2,2021,Spider-Man: No Way Home,5083.954,8940,8.3,Science Fiction
3,2022,The Batman,3827.658,1151,8.1,Crime
4,2022,The Batman,3827.658,1151,8.1,Mystery


## Column / Feature Analysis

In [ ]:
# Count frequency of each genre
df['Genre'].value_counts()

,count
Genre,
Drama,3743
Comedy,3030
Action,2683
Thriller,2485
Adventure,1852
Romance,1476
Horror,1469
Animation,1438
Family,1414


In [ ]:
# Dataset shape after preprocessing
df.shape

(25779, 6)

**Exploration Summary**

*   After completing the data cleaning and preprocessing steps, the dataset now contains 25,779 rows and 6 columns.
*   The dataset was generally well-structured, with only a small number of missing values and duplicate rows, which were handled during the data cleaning process.
*  The Release_Date column was converted to datetime format, and the year was extracted to support time-based analysis of movie releases.
*   Columns such as Overview, Poster_Url, and Original_Language were removed as they were not relevant for the intended numerical and categorical analysis.
*   The Genre column originally contained multiple genres in a single cell. It was split and expanded so that each genre appears in a separate row, enabling more detailed genre-level analysis.


## Data Visualization

###1️⃣ Which genre appears most often?


In [ ]:
# Count how many times each genre appears in the dataset
genre_counts = df['Genre'].value_counts()

# Create a bar chart showing the frequency of each genre
fig = px.bar(
    x=genre_counts.index,      # Genre names on X-axis
    y=genre_counts.values,     # Number of movies in each genre
    title="Most Common Movie Genres"
)

# Display the chart
fig.show()

Conclusion:
Drama is the most common genre in the dataset.

###2️⃣ Which movies have the highest popularity?

In [ ]:
# Sort movies by popularity in descending order and select top 10
top_movies = df.drop_duplicates(subset="Title") \
               .sort_values(by="Popularity", ascending=False) \
               .head(10)

# Create a horizontal bar chart for better readability of movie titles
fig = px.bar(
    top_movies,
    x="Popularity",        # Popularity score on X-axis
    y="Title",             # Movie titles on Y-axis
    orientation="h",       # Horizontal bars
    title="Top 10 Most Popular Movies"
)

# Display the chart
fig.show()

Conclusion:
Spider-Man: No Way Home has the highest popularity score in the dataset.
Other highly popular movies include The Batman, No Exit and Encanto.

###3️⃣ How has movie production changed over time?

In [ ]:
# Remove duplicate movies first (since each movie appears multiple times for different genres)
unique_movies = df.drop_duplicates(subset='Title')

# Count how many unique movies were released each year
movies_per_year = unique_movies['Release_Date'].value_counts().sort_index()

# Create a line chart to show movie production trend over time
fig = px.line(
    x=movies_per_year.index,    # Years
    y=movies_per_year.values,   # Number of movies released
    markers=True,               # Show points on the line
    title="Movies Released Over the Years"
)

# Display the chart
fig.show()

Conclusion:
The number of movies released has increased steadily over time, especially after the year 2000.
The slight drop after 2020 may be due to incomplete data for recent years or because newer movies are still being added to the dataset.

###4️⃣ Which genres have the highest average popularity?

In [ ]:
# Calculate the average popularity score for each genre
genre_popularity = df.groupby('Genre')['Popularity'].mean().reset_index()

# Create a bar chart showing average popularity of each genre
fig = px.bar(
    genre_popularity,
    x="Genre",        # Genre names
    y="Popularity",   # Average popularity score
    title="Average Popularity by Genre"
)

# Display the chart
fig.show()

Conclusion: Adventure movies have the highest average popularity,suggesting that audiences strongly engage with adventure-based content.
Other genres such as Action and Science Fiction also show relatively high popularity.

###5️⃣ What is the distribution of popularity scores?

In [ ]:
# Create a histogram to understand how popularity scores are distributed
fig = px.histogram(
    df,
    x="Popularity",      # Popularity values
    nbins=30,            # Number of bins in the histogram
    title="Distribution of Movie Popularity"
)

# Display the chart
fig.show()

Conclusion:
The popularity distribution is highly skewed.
Most movies are average.
Only a few movies become blockbuster hits.

Overall Insights

From the analysis of the Netflix movies dataset, several interesting trends can be observed.

• Drama is the most frequently occurring genre in the dataset, indicating that Netflix offers a large number of drama-based movies.

• Adventure movies show the highest average popularity, suggesting that audiences strongly engage with adventurous and exciting content.

• Genres such as Action and Science Fiction also receive high audience interest.

• The number of movie releases increased significantly after the year 2000, showing the rapid growth of the film industry and streaming platforms.

Overall, the analysis suggests that audiences tend to prefer entertaining and high-energy genres, particularly Action, Adventure, and Science Fiction.